python/cudf/cudf/core/column/decimal.py

```diff
@@ -0,0 +1,69 @@
+import more_itertools
+import pyarrow as pa
+
+from cudf import _lib as libcudf
+from cudf.core.buffer import Buffer
+from cudf.core.column import ColumnBase
+from cudf.core.dtypes import DecimalDtype
+from cudf.utils.utils import pa_mask_buffer_to_mask
+
+
+class DecimalColumn(ColumnBase):
+    @classmethod
+    def from_arrow(cls, data: pa.Array):
+        mask_buf = data.buffers()[0]
+        mask = (
+            mask_buf
+            if mask_buf is None
+            else pa_mask_buffer_to_mask(mask_buf, len(data))
+        )
+        bts = data.buffers()[1].to_pybytes()
+        bts = b"".join(list(more_itertools.sliced(bts, 8))[::2])
```

Comment:
why must half the bytes be discarded? Is this because you're going from `pa.decimal128` to `DECIMAL64`? Is this potentially lossy?

Reply:
It is potentially lossy-- but there's not much we can do about it since `pyarrow` has _only_ a `decimal128` type.

Reply:
Instead of numba we should be able to use numpy to create a strided array zero copy from the buffer and then make it contiguous if needed.

Reply:
Fixed.